std/math/exponential.d

```diff
@@ -3241,6 +3303,84 @@ private T logImpl(T)(T x) @safe pure nothrow @nogc
     return z;
 }
 
+@safe @nogc nothrow unittest
+{
+    import std.math : floatTraits, RealFormat;
+    import std.meta : AliasSeq;
+
+    static void testLog(T)(T[2][] vals)
+    {
+        import std.math.operations : isClose;
+        import std.math.traits : isNaN;
+        foreach (ref pair; vals)
+        {
+            if (isNaN(pair[1]))
+                assert(isNaN(log(pair[0])));
+            else
+                assert(isClose(log(pair[0]), pair[1]));
+        }
+    }
+    static foreach (F; AliasSeq!(float, double, real))
+    {{
+        F[2][24] vals = [
```

Comment:
Btw this is where new `scope F[2][] vals = …` should come in handy.

Reply:
In mainline, right?